apps/src/music/blockly/extensions.js

```diff
@@ -185,3 +188,28 @@ export const fieldSoundsValidator = function () {
     return newValue;
   });
 };
+
+/**
+ * Extension to blocks with pattern fields that validates new values.
+ */
+export const fieldPatternsValidator = function () {
+  // A block may have a pattern field or pattern AI field, but should not have both.
+  const patternField =
+    this.getField(FIELD_PATTERN_NAME) || this.getField(FIELD_PATTERN_AI_NAME);
+
+  /**
+   * Removes invalid event notes from pattern field values.
+   * @param newValue The new instrument event value
+   * @returns The modified instrument event value
+   */
+  patternField?.setValidator(newValue => {
+    newValue.events = newValue.events.filter(
+      event =>
+        // Remove events with notes that are outside the range of our supported drum sounds.
+        event.note < PATTERN_NUM_NOTES &&
```

Comment:
Optional; but since we have access to the field value (and presumably this will run only when the library has loaded), we could instead lookup the instrument in the library and just check that the note value matches one of the notes for the instrument?

Currently yes all drum sounds only have 8 sounds, and we assign notes to each sound based on the index (i.e. kick = 0, snare = 1), but if/when we use actual notes values (which I'd love to do soon after HOC), it would be great to not have a hardcoded assumption that we only support notes 0-7.

Reply:
Great idea, and thanks for the context. Updated!